File: batch/src/main/java/com/beat/batch/booking/job/TicketCleanupJob.java
package com.beat.batch.booking.job;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import com.beat.batch.booking.facade.TicketCleanupFacade;

import lombok.RequiredArgsConstructor;

@Component
@RequiredArgsConstructor
public class TicketCleanupJob {

	private final TicketCleanupFacade ticketCleanupFacade;

	@Value("${beat.scheduler.owner:false}")
	private boolean schedulerOwner;

	@Scheduled(cron = "0 0 4 * * ?")
	public void deleteOldCancelledBookings() {
		if (!schedulerOwner) {
			return;
		}

		ticketCleanupFacade.deleteOldCancelledBookings();
	}
}

File: batch/src/main/kotlin/com/beat/batch/booking/job/TicketCleanupJob.kt
package com.beat.batch.booking.job

import com.beat.application.system.booking.command.TicketCleanupService
import org.springframework.beans.factory.annotation.Value
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class TicketCleanupJob(
    private val ticketCleanupService: TicketCleanupService,
    @param:Value("\${beat.scheduler.owner:false}") private val schedulerOwner: Boolean,
) {
    @Scheduled(cron = "0 0 4 * * ?", scheduler = "maintenanceTaskScheduler")
    fun deleteOldCancelledBookings() {
        if (!schedulerOwner) return
        ticketCleanupService.deleteOldCancelledBookings()
    }
}
